### PR TITLE
Fix App crash when joing specific room URL

### PR DESCRIPTION
**Pull request by Freezlex**
Add `-` token to MATRIX_ROOM_IDENTIFIER_REGEX to prevent app from crashing when trying to create RoomId object.

A matrix room identifier can contain hyphens as in the identifier : `!gomuks-v10:maunium.net`

Parent issue :
 * #415

### CODE REVIEW

**Comment by Freezlex** [comment]
The Danger CI fail, does it need something ?
Seems to be due to the fact that the PR is from a fork.

File: libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/core/MatrixPatterns.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ object MatrixPatterns {
     val PATTERN_CONTAIN_MATRIX_USER_IDENTIFIER = MATRIX_USER_IDENTIFIER_REGEX.toRegex(RegexOption.IGNORE_CASE)
 
     // regex pattern to find room ids in a string.
-    private const val MATRIX_ROOM_IDENTIFIER_REGEX = "![A-Z0-9]+$DOMAIN_REGEX"
+    private const val MATRIX_ROOM_IDENTIFIER_REGEX = "![A-Z0-9-]+$DOMAIN_REGEX"
```
[the remark this answers is not in the document]
Comment (Freezlex) [reply libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/core/MatrixPatterns.kt:37]: I can try on Element Android, I haven't tried it yet but as far as I remember, I never had this problem before on Android Element even if the Regex failed (obviously if it's the same). I'll let you know when I've tried it.

[the remark this answers is not in the document]
Comment (Freezlex) [reply libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/core/MatrixPatterns.kt:37]: > This code is coming from [Element Android](https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixPatterns.kt#L39). I guess we have some issue there as well.

The issue does not appear in Element Android, it seems that the problem is better handled.

**Comment by bmarty** [comment]
@Freezlex why did you closed your PR?

**Comment by Freezlex** [comment]
> @Freezlex why did you closed your PR?

I thought from your previous reply that this was an irrelevant bug or related to Element Android.

**Comment by bmarty** [comment]
Thanks @Freezlex . Can you add a sign-off so that we can merge the PR? Thanks!

**Comment by Freezlex** [comment]
Sign-off sent to dco@matrix.org

**Review by bmarty — APPROVED** [review]
Thanks!

**Comment by InezMc** [comment]
Anonymous Sign-off recieved and recorded from Freezlex.

**Pull request merged**